R/fct_SQLite.R

```diff
@@ -164,6 +164,16 @@ db_add_log <- function(con) {
                        edit_date_time = "CHAR", reviewed = "CHAR", comment = "CHAR", 
                        reviewer = "CHAR", timestamp = "CHAR", status = "CHAR",
                        dml_type = "CHAR NOT NULL", dml_timestamp = "DATETIME DEFAULT CURRENT_TIMESTAMP"))
+  # This will trigger before any UPDATEs happen on all_review_data. Instead of
+  # allowing 'id' to be updated, it will throw an error.
+  rs <- DBI::dbSendStatement(con, paste(
+    "CREATE TRIGGER all_review_data_id_update_trigger",
+    "BEFORE UPDATE OF id ON all_review_data",
+    "BEGIN",
+    "SELECT RAISE(FAIL, 'all_review_data.id is read only');",
+    "END"
+  ))
+  DBI::dbClearResult(rs)
```

Comment:
I tried to test this behavior with a test database. Could it be that these are not updated yet? Also, is it possible to capture this in a test? It is quite important that this does not go wrong.

Reply:
Should this behavior be extended to all `idx_cols`?

Reply:
Good question. I feel like the id column is the most important since it links the review data to the audit/log table. 

I am trying to thing if there would be a moment that one of the idx_cols/common_vars gets updated. 
What would happen if this occurs for the event_name (e.g. from `visit 1` to `visit 2`, although I think that would be rare)? Would that count as a completely new row with a new id?

Reply:
Well with the construction as it stands now (and how it functioned previously I believe) this would get input to the review data base as a new record and a warning would be triggered indicating "items were not found in the updated dataset". See `update_review_data()`.